src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -339,6 +340,9 @@ def analyze(
                 sql = sql.upper()
             return sql
 
+        if isinstance(expr, Interval):
```

Comment:
The local test code also needs to add Interval. cc @sfc-gh-stan, @sfc-gh-aling

Reply:
@sfc-gh-yixie Sorry not very familiar with this codebase, can you clarify what other tests need to be added?

Reply:
It's under `snowflake.snowpark.mock`. Snowpark has mock to help users test their code without connecting to Snowflake. It simulate some snowflake behavior locally so need to handle the new Interval expression. Interval doesn't have a public API yet so I think it's OK not to do it in the mock at this time. But needs @sfc-gh-stan and / or @sfc-gh-aling to sign off.

Reply:
thanks, logged a JIRA for this: https://snowflakecomputing.atlassian.net/browse/SNOW-1005178